### ISSUE TITLE
[feature request] More flexible unsubscribe functionality

### ISSUE THREAD

**Issue by that0n3guy** [mautic/mautic#822]
Right now, unsubscribes are global (meaning leads are unsubscribed from all emails).  Below are my feature requests:
### Unsubscribe per email & per list, not per lead

**Per email**
If a lead has multiple email addresses (may depend on #821), often times they will want to unsubscribe from one, but not the other (work, but not personal).

Also, if a user unsubscribes then we manually update the email address to a new one (per customers/leads request), they are still unsubscribed.

**Per list (or list category)**
Unsubscribe per list is pretty industry standard for email marketing software.  

We have several lists.  Some lists are more targeted to a specific product types.  Sometimes a user wants to unsubscribe from getting info about productX but not productY... mautic has no way of handling this now.
### Global unsubscribe

This is also an industry standard.  You see a lot of unsubscribe pages that allow the user to unsubscribe from all lists.

This is not the same as a suppression list (https://github.com/mautic/mautic/issues/820).  A the ability to unsubscribe globally unsubscribes the email address from all current lists.  It does not unsubscribe them from any they may re-subscribe to in the future.

A suppression list blocks their email from EVER getting an email.

**Comment by npracht** [comment]
Hi,

This is a pretty great feature -and you're right- used in most of the Marketing Automation softwares.
They use the term of "Subscription List Management".
To complete your approach of product A/B/C. You can also for a company have more than one brand.

**Comment by mscotto** [comment]
Mautic is awesome, thanks for all the amazing work. I agree a more granular approach to unsubscribe is needed. Digging into the code and db schema it appears that the attaching unsubscribe to the list would require a bit of work both UX and in code. The main problem being that the emails do not always source from a list.

So instead a possible simple but effective fix would be to attached unsubscribes to both the receiver email and sender email. This way regardless of the email source, list or form, email subscription could be linked together by the sender email. In the short term it could be done with no UI changes, possible one setting to select unsubscribe Global vs By Sender.

**Comment by dbarrow1992** [comment]
I would like to say this is some good thinking! Sadly i cant give any technical support. I'd love to see these as a function in the next iterations of Mautic!

**Comment by OlafDammann** [comment]
Yes, a Self Service Portal for the users is a great idea :)
+1

**Comment by that0n3guy** [comment]
I'd love to see unsubscribe per list in 2.0...

**Comment by ipso** [comment]
Yes, this feature is definitely needed!

**Comment by ninjoan** [comment]
definitely needed

**Comment by zstephen** [comment]
What can we do to get some momentum behind this feature request? I can't write code, but surely I can help some other way

**Comment by that0n3guy** [comment]
Tomorrow will be the 1 year anniversary of this issue :).  I thought I would post another idea:

## Unsubscribe categories. 

This is a balance between the global unsubscribe and the per list/segment unsubscribe.   It's where you create several global categories for emails (Newsletter, Promotions, Product X category, Updates, etc...) and the user can unsubscribe from a category rather than a list or globally.

**How does this method work:**  When sending an email inside mautic, you would select what "Unsubscribe category" the email belongs to. When a user unsubscribes, they are unsubscribed from the sent category and would no longer be sent emails in future emails of that category (unless they re-subscribe).

Additional Thoughts:

- Act-on uses this method (or you can use the global method) instead of unsubscribing on a per segment basis.
- I realize emails already have a category on them, but I don't believe this should be used for the "unsubscribe category".   Current categories are for backend organization and may be titled something odd.    Unsubscribe categories are more what the contact might expect.

**Comment by zstephen** [comment]
thanks for sharing this suggestion @that0n3guy

**Comment by that0n3guy** [comment]
A work-around for this: https://medium.com/@NinJoan/simplest-way-to-make-a-unsubscribe-system-for-mautic-4f307850cce2

The "unsubscribe categories" I posted about could probably be done with this method as well. It might make list segments and managing a little more tricky, but still doable.

**Comment by ninjoan** [comment]
Cool! thanks

**Comment by that0n3guy** [comment]
I should note, that while ninjoan's solutions works, it is just a work around.

It does not provide:

- reporting on which email was sent that caused the person to unsubscribe
- reporting on number of unsubscribes per email

I still feel that a core implementation is still required.

**Comment by magician7** [comment]
Definitely options to simply insert in mail unsubscribe link from only category / segment is needed. From core not from work around.

**Comment by merlin0** [comment]
+1 We've been using a custom work around but since we've been adding campaigns it gets harder to implement and it would be a huge time saver.

**Comment by escopecz** [comment]
Preference center added in Mautic 2.3

**Comment by npracht** [comment]
Hi @escopecz,

I do not agree. The ability of managing different subscription lists of contacts is not part of the preference center. Here is the description, maybe @mqueme can confirm it is not the same:

## Subscription List Management
Contacts can be interested by different type of content given by the company. Product A/B/C, more than one brand, newsletter, events, etc.

### How: create Subscription List
Like different categories and generate different unsubscribe tokens:
* Newsletter --> `{unsubscribe_url=newsletter}`
* Updates --> `{unsubscribe_url=updates}`
* Products --> `{unsubscribe_url=products}`
* Events --> `{unsubscribe_url=events}`
* Brand A --> `{unsubscribe_url=branda}`
* Brand B --> `{unsubscribe_url=brandb}`

### What then happens?

1. It'll unsubscribe the contact about the topic he clicked on. But not the others.
2. You'd still have the ability to insert global token if wished.
3. You can imagine also another token like `{unsubscribe_url=preferences}` that the contact can click on to see all the Subscription Lists available and where he is OFF/ON.
__

**Does preference center brings that feature ?**
Thanks for reading !

N.

**Comment by escopecz** [comment]
@npracht Let's leave this issue closed because "More flexible unsubscribe functionality" = "Preference center". If you'd like the special unsubscribe links, please create a new issue for it. It's really hard to track issues done from 90%.

**Comment by npracht** [comment]
clear !

**Comment by mqueme** [comment]
The new preference center has a 'My categories' sections where users can unsubscribe from categories. the preference center gives the user the ability to unsubscribe but not to subscribe to categories, segments and to subscribe/unsubscribe to channels

**Comment by stylefieber** [comment]
It's a good step forward, but I still miss multiple contact-lists which behave completely independently from each other.

So that you are able to host multiple different projects with their customers / contacts.

Currently, you need one instance of Mautic for every independent List of contacts.
